src/shared/particle_dynamics/continuum_dynamics/continuum_integration.cpp

```diff
@@ -116,14 +135,116 @@ void StressDiffusion::interaction(size_t index_i, Real dt)
+}
+//====================================================================================//
+void ShearStressRelaxationHourglassControl::update(size_t index_i, Real dt)
+{
+    Real rho_i = rho_[index_i];
+    Matd shear_stress_i = shear_stress_[index_i];
+    Vecd acceleration = Vecd::Zero();
+    Vecd acceleration_hourglass = Vecd::Zero();
+    Neighborhood &inner_neighborhood = inner_configuration_[index_i];
+    for (size_t n = 0; n != inner_neighborhood.current_size_; ++n)
+    {
+        size_t index_j = inner_neighborhood.j_[n];
+        Real dW_ijV_j = inner_neighborhood.dW_ij_[n] * Vol_[index_j];
+        Vecd &e_ij = inner_neighborhood.e_ij_[n];
+        acceleration += ((shear_stress_i + shear_stress_[index_j]) / rho_i) * dW_ijV_j * e_ij;
+        Vecd v_ij = vel_[index_i] - vel_[index_j];
+        Real r_ij = inner_neighborhood.r_ij_[n];
+        Vecd v_ij_correction = v_ij - 0.5 * (velocity_gradient_[index_i] + velocity_gradient_[index_j]) * r_ij * e_ij;
+        acceleration_hourglass += 0.5 * xi_ * (scale_penalty_force_[index_i] + scale_penalty_force_[index_j]) * G_ * v_ij_correction * dW_ijV_j * dt / (rho_i * r_ij);
+    }
+    acc_hourglass_[index_i] += acceleration_hourglass;
+    acc_shear_[index_i] = acceleration + acc_hourglass_[index_i];
+}
+//====================================================================================//
+ShearStressRelaxationHourglassControlJ2Plasticity ::
+    ShearStressRelaxationHourglassControlJ2Plasticity(BaseInnerRelation &inner_relation, Real xi)
+    : ShearStressRelaxationHourglassControl(inner_relation, xi),
+      J2_plasticity_(DynamicCast<J2Plasticity>(this, particles_->getBaseMaterial())),
+      hardening_factor_(particles_->registerStateVariable<Real>("HardeningFactor"))
+{
+    particles_->addVariableToSort<Real>("HardeningFactor");
+}
+//====================================================================================//
+void ShearStressRelaxationHourglassControlJ2Plasticity::interaction(size_t index_i, Real dt)
+{
+    shear_stress_rate_[index_i] = J2_plasticity_.ConstitutiveRelationShearStress(velocity_gradient_[index_i], shear_stress_[index_i], hardening_factor_[index_i]);
+    Matd shear_stress_try = shear_stress_[index_i] + shear_stress_rate_[index_i] * dt;
+    Real hardening_factor_increment = J2_plasticity_.HardeningFactorRate(shear_stress_try, hardening_factor_[index_i]);
+    hardening_factor_[index_i] += sqrt(2.0 / 3.0) * hardening_factor_increment;
+    scale_penalty_force_[index_i] = J2_plasticity_.ScalePenaltyForce(shear_stress_try, hardening_factor_[index_i]);
+    shear_stress_[index_i] = J2_plasticity_.ReturnMappingShearStress(shear_stress_try, hardening_factor_[index_i]);
+    Matd stress_tensor_i = shear_stress_[index_i] - p_[index_i] * Matd::Identity();
+    von_mises_stress_[index_i] = getVonMisesStressFromMatrix(stress_tensor_i);
+    Matd strain_rate = 0.5 * (velocity_gradient_[index_i] + velocity_gradient_[index_i].transpose());
```

Comment:
'von_mises_stress_' should not be stored in the memory. We can calculate it at the case main function, e.g., 'SimpleDynamics<VonMisesStress> plate_stress(plate)'. Same for 'von_mises_strain_'

Reply:
Thanks! I will revise this.